testcases/open_posix_testsuite/conformance/interfaces/strcpy/1-1.c

```diff
@@ -0,0 +1,54 @@
+/*
+ * This file is licensed under the GPL license. For the full content
+ * of this license, see the COPYING file at the top level of this
+ * source tree.
+ */
+
+/*
+ * assertion:
+ *	The function will copy the string pointed to by s1 (including the
+ *	terminating NUL character) into the array pointed to by s2. And
+ *	it will return s2.
+ *
+ * method:
+ *	-Define a sample string s1.
+ *	-Use strcpy() to copy s1 into sample string s2.
+ *	-Compare both strings(s1 and s2).
+ *	-Also compare returned pointer with s2.
+*/
+
+#define _XOPEN_SOURCE 700
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include "posixtest.h"
+
+#include "../testfrmw/testfrmw.h"
+#include "../testfrmw/testfrmw.c"
+
+#define TNAME "strcpy/1-1.c"
+
+int main(void)
+{
+	static char sample_str_1[] = "HELLO WORLD";
+	char sample_str_2[sizeof(sample_str_1)] = { };
+	char *ret_str = NULL;
+
+	ret_str = strcpy(sample_str_2, sample_str_1);
+
+	if (strcmp(sample_str_1, sample_str_2) == 0 && ret_str == sample_str_2) {
+		output(TNAME " Test Passed, string copied successfully\n");
+		exit(PTS_PASS);
+	} else if (strcmp(sample_str_1, sample_str_2) != 0) {
+		output(TNAME " Test Failed, string copy failed. "
+					"Expected string: %s, But got: %s\n"
+					, sample_str_1, sample_str_2);
+		exit(PTS_FAIL);
+	} else {
+		output(TNAME " Test Failed, return is not as expected. "
+					"Expected return: %p, But obtained: %p\n"
+					, sample_str_2, ret_str);
+		exit(PTS_FAIL);
+	}
```

Comment:
There is no need for the else branches since you call exit() in the if body, so this could be simplified to:

```
        if (foo) {
                ...
                exit(...);
        }

        if (bar) {
                 ...
                 exit(...);
        }

        ...
```

Reply:
I think code will be more readable with if-else.
if (1 && 2) {
}
else if ( !1 ) {
}
else { // !2
}